From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Wed, 19 Feb 2020 11:00:00 +0100
Subject: [PATCH 1/1] Add --disable-cpp option

Handy for building with -fno-exceptions.

Upstream-Status: Inappropriate [disable feature]
This patch is just for our convenience.

diff --git a/Makefile.am b/Makefile.am
index 1111111..2222222 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,16 @@
+if ENABLE_CPP
+CPP_COMPILE_DIR = cplusplus
+CPP_PKG = vips-cpp.pc
+CPP_PKG_IN = vips-cpp.pc.in
+else
+CPP_COMPILE_DIR =
+CPP_PKG =
+CPP_PKG_IN =
+endif
+
 SUBDIRS = \
 	libvips \
-	cplusplus \
+	$(CPP_COMPILE_DIR) \
 	tools \
 	po \
 	man \
@@ -13,13 +23,13 @@ EXTRA_DIST = \
 	benchmark \
 	autogen.sh \
 	vips.pc.in \
-	vips-cpp.pc.in \
+	$(CPP_PKG_IN) \
 	suppressions \
 	depcomp \
 	README.md 
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = vips.pc vips-cpp.pc
+pkgconfig_DATA = vips.pc $(CPP_PKG)
 
 dist-hook:
 # make sure we don't get any .svn dirs from EXTRA_DIST
diff --git a/configure.ac b/configure.ac
index 1111111..2222222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1265,6 +1265,13 @@ if test x"$with_libexif" != x"no"; then
   CPPFLAGS="$save_CPPFLAGS"
 fi
 
+# option to disable the cpp binding, handy for building with -fno-exceptions
+AC_ARG_ENABLE([cpp], 
+  AS_HELP_STRING([--disable-cpp], 
+		 [disable cpp binding (default: enabled)]))
+
+AM_CONDITIONAL(ENABLE_CPP, [test x"$enable_cpp" != x"no"])
+
 # fuzzing
 AC_ARG_VAR([LIB_FUZZING_ENGINE],
            [fuzzing library, e.g. /path/to/libFuzzer.a])
